config/crd/bases/kuadrant.io_dnspolicies.yaml

```diff
@@ -0,0 +1,156 @@
+---
+apiVersion: apiextensions.k8s.io/v1
+kind: CustomResourceDefinition
+metadata:
+  annotations:
+    controller-gen.kubebuilder.io/version: v0.10.0
+  creationTimestamp: null
+  name: dnspolicies.kuadrant.io
+spec:
+  group: kuadrant.io
+  names:
+    kind: DNSPolicy
+    listKind: DNSPolicyList
+    plural: dnspolicies
+    singular: dnspolicy
+  scope: Namespaced
+  versions:
+  - additionalPrinterColumns:
+    - description: DNSPolicy ready.
+      jsonPath: .status.conditions[?(@.type=="Ready")].status
+      name: Ready
+      type: string
+    name: v1alpha1
+    schema:
+      openAPIV3Schema:
+        description: DNSPolicy is the Schema for the dnspolicies API
+        properties:
+          apiVersion:
+            description: 'APIVersion defines the versioned schema of this representation
+              of an object. Servers should convert recognized schemas to the latest
+              internal value, and may reject unrecognized values. More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#resources'
+            type: string
+          kind:
+            description: 'Kind is a string value representing the REST resource this
+              object represents. Servers may infer this from the endpoint the client
+              submits requests to. Cannot be updated. In CamelCase. More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#types-kinds'
+            type: string
+          metadata:
+            type: object
+          spec:
+            description: DNSPolicySpec defines the desired state of DNSPolicy
+            properties:
+              dnsRecordRef:
```

Comment:
Two thoughts here:

1) I had thought we would be targeting a gateway with this API and the controller would then translate the check into DNSRecord changes
2) with the targeting, I think we should reuse the targetRef concept from policy attachment and gateway api 

```  
  targetRef: # This struct is defined as part of Gateway API
    group: 
    kind: 
    name: 
 ```
https://gateway-api.sigs.k8s.io/references/policy-attachment/

Reply:
also I thik we only support targeting resources in the same ns

Reply:
I see. I had misunderstood and thought we were moving away from directly referencing the gateway in the policy. So I guess the controller when reconciling gets the Gateway and then the associated DNSRecords from it?

Reply:
yes so each listener in a gateway will have a DNSRecord so controller can find that DNSRecord and apply the health check.

Reply:
Should we try and keep this consistent with how the RLP does this and use `targeRef` with `group`, `kind`, `name` and as @maleck13 says it will be in teh same ns?